examples/cache.ex

```diff
@@ -0,0 +1,76 @@
+defmodule ExampleSupervisor do
+  def start do
+    import Supervisor.Spec
+
+    # List comprehension creates a consumer per cpu core
+    children = for i <- 1..System.schedulers_online, do: worker(ExampleConsumer, [], id: i)
+
+    Supervisor.start_link(children, strategy: :one_for_one)
+  end
+end
+
+defmodule ExampleConsumer do
+  use Nostrum.Consumer
+
+  alias Nostrum.Api
+
+  require Logger
+
+  def start_link do
+    Consumer.start_link(__MODULE__, :state)
+  end
+
+  def handle_event({:MESSAGE_CREATE, {msg}, ws_state}, state) do
+    # Let's do a little command 'parsing'
+    # We first grab the content, split it into an array and then convert the array to a tuple
+    args = msg.content |> String.split(" ") |> List.to_tuple
+
+    case elem(args, 0) do # We check the very first arg to see the command
+      "!userinfo" -> # !userinfo user_id
+        if tuple_size(args) == 2 do # Check to make sure there are 2 arguments, meaning the command, and then an argument
+
+          # We grab the second thing from the tuple
+          user_id = elem(args, 1)
+
+          if user_id != nil do
+            try do
+              # try and convert the supplied input to an integer
+              user_id = user_id |> String.to_integer
```

Comment:
I'd recommend we use `Integer.parse/1` here. By doing so we could remove the try/rescue and flatten the `with` statement even further.